server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -217,11 +223,13 @@ protected static boolean parseObjectOrDocumentTypeProperties(
                 }
                 return true;
             } else if (fieldName.equals("include_in_all")) {
-                deprecationLogger.warn(
-                    DeprecationCategory.MAPPINGS,
-                    "include_in_all",
-                    "[include_in_all] is deprecated, the _all field have been removed in this version"
-                );
+                if (parserContext.indexVersionCreated().isLegacyIndexVersion() == false) {
```

Comment:
Do we need to guard this with a version check? I think it will still be true even for older mappings, in that we won't generate an `_all` field type that you can search against even if the underlying lucene field exists in the index?

Reply:
Good spot. In an earlier iteration, I was pondering on adding _all support for legacy indices, hence the version check. I've removed it now.